front/src/features/books/booksSlice.tsx

```diff
@@ -0,0 +1,81 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+import api from "api/index";
+import { initialState } from "./helperData";
+
+import { IfetchBooksData, IAddItemToBasket } from "../interfaces/index";
+
+export const booksSlice = createSlice({
+  name: "books",
+  initialState: initialState,
+  reducers: {
+    fetchBooksDataStarted: (state) => {
+      state.status = "inProgress";
+    },
+    fetchBooksDataSucceeded: (
+      state,
+      action: PayloadAction<IfetchBooksData>
+    ) => {
+      state.data = [...action.payload.data];
+      state.status = "succeeded";
+    },
+    fetchBooksDataFailed: (state) => {
+      state.status = "failed";
+    },
+    addItemToBasket: (state, action: PayloadAction<IAddItemToBasket>) => {
```

Comment:
to nie będzie IAddItemToBasket, tylko po prostu BasketItem. Jeżeli byłoby tam więcej info, niż sam item, to mógłbyś zrobić coś pokroju IAddItemToBasketPayload = {
basketItem: BasketItem
cośtam: string
costaminneg?: boolean
}

Reply:
Done.